### PR TITLE
Add NTLM feature detection and more tests

### PR DESCRIPTION
**Pull request by davidsh**
This PR works toward the goal of our Enterprise Scenarios. I added logic in the Unix
PAL gssapi layer to detect whether the NTLM plugin is installed. This is useful both
in the product code (to return better exception messages) and the test code (to
skip NTLM tests if needed).

Refactored some of the Common System.Net.Configuration logic for tests. Added new
environment variable logic to separate between domain-joined and standalone server
environments.

Add new tests to validate NTLM end-to-end connectivity. Opened some new issues
discovered along with way regarding HTTP/2 and Windows authentication.

To run these new tests simply set the 3 environment variables to point to the
standalone windows server environment.  For example, on Linux:

```
export COREFX_WINDOWSSERVER_HTTPHOST=somewindowsmachine.contoso.com
export COREFX_NET_SERVER_USERNAME=someusername
export COREFX_NET_SERVER_PASSWORD=somepassword
```

I have already deployed a standalone Windows server for these tests but
we're not ready yet to have the tests always run in CI.

Contributes to #34878

### CODE REVIEW

**Comment by stephentoub** [comment]
@dotnet-bot test this please

File: src/System.Net.Http/tests/FunctionalTests/System.Net.Http.Functional.Tests.csproj [hunk 1]
```diff
@@ -40,6 +43,12 @@
     <Compile Include="$(CommonTestPath)\System\Net\Capability.Security.cs">
       <Link>Common\System\Net\Capability.Security.cs</Link>
     </Compile>
+    <Compile Include="$(CommonTestPath)\System\Net\Capability.Security.Windows.cs" Condition="'$(TargetsWindows)' == 'true'">
+      <Link>Common\System\Net\Capability.Security.Windows.cs</Link>
+    </Compile>
```
Comment (stephentoub) [inline src/System.Net.Http/tests/FunctionalTests/System.Net.Http.Functional.Tests.csproj]: This file appears to be missing.

Comment (davidsh) [reply src/System.Net.Http/tests/FunctionalTests/System.Net.Http.Functional.Tests.csproj]: Thanks.  I'm still learning about command line git versus the easy Git Desktop for Windows which moves all files into a commit.

**Comment by davidsh** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build
@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop NETFX x86 Debug Build
@dotnet-bot test Outerloop UWP CoreCLR x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Debug Build

**Comment by davidsh** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build
@dotnet-bot test Outerloop NETFX x86 Debug Build
@dotnet-bot test Outerloop UWP CoreCLR x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Debug Build

File: src/Common/src/Microsoft/Win32/SafeHandles/GssSafeHandles.cs [hunk 2]
```diff
@@ -78,6 +78,14 @@ internal class SafeGssCredHandle : SafeHandle
         /// </summary>
         public static SafeGssCredHandle Create(string username, string password, bool isNtlmOnly)
         {
+            if (isNtlmOnly && !Interop.NetSecurityNative.IsNtlmInstalled())
```
Comment (stephentoub) [inline src/Common/src/Microsoft/Win32/SafeHandles/GssSafeHandles.cs]: Can the IsNtlmInstalled call be cached?

File: src/Native/Unix/System.Net.Security.Native/pal_gssapi.c [hunk 3]
```diff
@@ -427,3 +427,35 @@ uint32_t NetSecurityNative_InitiateCredWithPassword(uint32_t* minorStatus,
...
+    majorStatus = gss_indicate_mechs(&minorStatus, &mechSet);
+    if (majorStatus == GSS_S_COMPLETE)
+    {
+        for (size_t i = 0; i < mechSet->count; i++)
+        {
+            oid = mechSet->elements[i];
+            if ((oid.length == ntlmOid->length) && (memcmp(oid.elements, ntlmOid->elements, oid.length) == 0))
+            {
+                foundNtlm = 1;
+            }
```
Comment (stephentoub) [inline src/Native/Unix/System.Net.Security.Native/pal_gssapi.c]: Can we break out of the loop as well?

Comment (davidsh) [reply src/Native/Unix/System.Net.Security.Native/pal_gssapi.c]: Yes.  I forgot the `break` statement.  I'll fix that.

File: src/Common/src/Microsoft/Win32/SafeHandles/GssSafeHandles.cs (hunk 2 above)
Comment (davidsh) [reply src/Common/src/Microsoft/Win32/SafeHandles/GssSafeHandles.cs]: Yes.  I can probably cache this at the .NET layer with a Lazy<T>.

**Comment by davidsh** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build
@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop NETFX x86 Debug Build
@dotnet-bot test Outerloop UWP CoreCLR x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Debug Build

**Comment by davidsh** [comment]
Failures in CI not related to PR.

**Pull request merged**